### ISSUE TITLE
 fix cctest test-assembler-riscv32/RISCV_UTEST_mul{h/s/u}

### ISSUE THREAD

**Issue by luyahan** [riscv-collab/v8#594]
```
 30 Command: out/riscv32.debug/cctest test-assembler-riscv32/RISCV_UTEST_mulhsu --random-seed=-75565327 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
 31 Command: out/riscv32.debug/cctest test-assembler-riscv32/RISCV_UTEST_mulh --random-seed=-75565327 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
 32 Command: out/riscv32.debug/cctest test-assembler-riscv32/RISCV_UTEST_mulhu --random-seed=-75565327 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
````

**Comment by luyahan** [comment]
@usmnzain can you fix these three error?

**Comment by qjivy** [comment]
```
./out/riscv32.debug/cctest test-assembler-riscv32/RISCV_UTEST_mulhsu --random-seed=-75565327 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner^C
qjivy@p9-plct:/work/qjivy/rv32v8/another/v8$ out/riscv32.debug/cctest test-assembler-riscv32/RISCV_UTEST_mulhsu --random-seed=-75565327 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner


#
# Fatal error in ../../test/cctest/test-assembler-riscv32.cc, line 394
# Check failed: static_cast<int32_t>((-0x12345678LL * 0xF2345678ULL) >> 32) == res (-288961158 vs. -502585408).
#
#
#
#FailureMessage Object: 0xffb6d2b8
```

The result of mulhsu for (-0x12345678LL * 0xF2345678ULL) is not correct, we should check if the simulator is not correct for this instruction.
Seems as simulator error.

**Comment by usmnzain** [comment]
@luyahan I will look into these errors

**Comment by luyahan** [comment]
```
abort: CSA_DCHECK failed: Torque assert '0 <= input && input < 36' failed [src/builtins/number.tq:65] [../../src/builtins/number.tq:106] [../../src/builtins/cast.tq:72] [../../src/builtins/array-join.tq:443] [../../src/builtins/array-join.tq:572] [../../src/builtins/array-join.tq:658]
```
The errors relate with this issue.

**Comment by usmnzain** [comment]
Fixed in PR #614
